experimental/piranha_playground/rule_inference/tests/test_piranha_agent.py

```diff
@@ -40,13 +40,41 @@ def test_go_inference():
     }
     """
 
+    # Initialize parser and parse the code
+    agent = PiranhaAgent(source_code, target_code, language)
+    res = agent.infer_rules_statically()
+    graph = RawRuleGraph.from_toml(toml.loads(res))
+    expected = [
+        "((short_var_declaration",
+        " left: (expression_list ",
+        "  (identifier ) @tag3n) @tag2n",
+        " right: (_) @tag4n) @tag1n",
+        '(#eq? @tag3n "age"))',
+    ]
+    assert len(graph.rules) == 1 and graph.rules[0].query == "\n".join(expected)
+
+
+def test_java_inference():
+    # Source and target code samples
+    language = "java"
+    source_code = """ // 1
+    :[x].doSomething(:[y]); 
+    """
+    target_code = """ // 1
```

Comment:
target code is `after` right? Why do we have to give the "kind" list in rhs and not in LHS

Reply:
You can do both, only need to do it once

Reply:
I see. Let's still have it on LHS (in this test) . It is more intuitive :|